src/track/steminfoimporter.cpp

```diff
@@ -0,0 +1,144 @@
+        uint32_t boxSize = 0,
+        uint32_t pathIdx = 0) {
+    if (!path[pathIdx]) {
+        return boxSize;
+    }
+
+    uint32_t byteRead = 0;
+    char buffer[kAtomHeaderSize];
+    while (!reader.atEnd() || (boxSize && byteRead >= boxSize)) {
+        reader.read(buffer, kAtomHeaderSize);
+        byteRead += boxSize;
+        if (ATOM_TYPE(buffer + 4) == path[pathIdx]) {
+            return seekTillAtom(reader, path, ATOM_TYPE(buffer) - kAtomHeaderSize, pathIdx + 1);
+        }
+        reader.skip(ATOM_TYPE(buffer) - kAtomHeaderSize);
+    }
+    return -1;
+}
+
+} // anonymous namespace
+
+// static
+bool StemInfoImporter::isStemFile(
+        const QString& fileName) {
+    for (const QString& ext : kStemFileExtensions) {
+        if (fileName.endsWith(ext)) {
+            return true;
+        }
+    }
+    return false;
+}
+
+QList<StemInfo> StemInfoImporter::importStemInfos(
+        const QString& filePath) {
+    // Fetch the STEM manifest which contain stream details
+    auto file = QFile(filePath);
+    if (!file.open(QIODeviceBase::ReadOnly)) {
+        kLogger.warning()
+                << "Failed to open input file"
+                << filePath;
+        return {};
+    }
+
+    uint32_t manifestSize;
+    if (!(manifestSize = seekTillAtom(file, kStemManifestAtomPath))) {
+        kLogger.warning()
+                << "Failed to find the steam manifest in the file"
+                << filePath;
+        return {};
+    }
+
+    auto jsonData = QJsonDocument::fromJson(file.read(manifestSize));
+    VERIFY_OR_DEBUG_ASSERT(jsonData.isObject()) {
+        kLogger.warning()
+                << "Failed to extract the manifest data"
+                << filePath;
+        return {};
+    };
+
+    auto json = jsonData.object();
+
+    // Check the manifest version
+    auto version = json.value("version").toInteger(-1);
+    if (version <= 0) {
+        kLogger.debug() << "Unexpected or missing version value in STEM manifest";
+        return {};
+    } else if (version > kSupportedStemVersion) {
+        kLogger.debug() << "Unsupported stem version" << version << "but trying anyway";
+    }
+
+    // Extract stem metadata
+    auto stems = json.value("stems");
+    if (!stems.isArray()) {
+        kLogger.debug() << "Unexpected or missing stems value in STEM manifest";
+        return {};
+    }
+    auto stemArray = stems.toArray();
+    QList<StemInfo> stemsList;
+    stemsList.reserve(stemArray.size());
+    for (const auto& stemRef : stemArray) {
+        if (!stemRef.isObject()) {
+            kLogger.debug() << "Unexpected or missing stems value in STEM manifest";
+            return {};
+        }
+        auto stem = stemRef.toObject();
+        auto color = QColor(stem.value("color").toString());
+        auto name = stem.value("name").toString();
+        if (!color.isValid() || name.isEmpty()) {
+            kLogger.debug() << "Unexpected or missing stem name or attribute in STEM manifest";
+            return {};
```

Comment:
I think we should be less critical and pick reasonable defaults. A stem file should be play-able, with a minimum of information that is required to play it.

Reply:
"Stem #<n>" and Red/Green/Blue/Yellow?

Reply:
Not entirely pure, but some red, green and blue are used by default for waveform as mentioned in my [3-band waveform PR](https://github.com/mixxxdj/mixxx/pull/13075)

Reply:
Let's use the ones from stemgen, as the practical use case is stem splitting with stemgen and than use the generated files in Mixxx.